drivers/adc/adc_stm32.c

```diff
@@ -758,71 +909,84 @@ static int start_read(const struct device *dev,
 		return -EINVAL;
 	}
 
-	uint32_t channels = sequence->channels;
-	uint8_t index = find_lsb_set(channels) - 1;
-
-	if (channels > BIT(index)) {
-		LOG_ERR("Only single channel supported");
-		return -ENOTSUP;
-	}
+	data->buffer = sequence->buffer;
+#ifdef CONFIG_ADC_STM32_DMA
+	data->buffer_size = sequence->buffer_size;
+#endif
+	data->channels = sequence->channels;
+	data->channel_count = 0;
+	data->samples_count = 0;
 
-	adc_stm32_setup_channels(dev, index);
+	for (uint8_t channel_id = 0; channel_id < STM32_CHANNEL_COUNT; channel_id++) {
+		if (data->channels & BIT(channel_id)) {
+			uint32_t channel = __LL_ADC_DECIMAL_NB_TO_CHANNEL(channel_id);
```

Comment:
_else_ we have no channel value, is it sure that at least one `data->channels & BIT(channel_id))` will be true ?

Reply:
It's checked after the `for` loop if there were channels selected or not.
```C
	if (data->channel_count == 0) {
		LOG_ERR("No channels selected");
		return -EINVAL;
	}
```